spring-integration-jms/src/main/java/org/springframework/integration/jms/PollableJmsChannel.java

```diff
@@ -123,21 +125,13 @@ private Message<?> receiveAndConvertToMessage() {
 			}
 			return null;
 		}
-		else {
-			Message<?> message;
-			if (object instanceof Message<?>) {
-				message = (Message<?>) object;
-			}
-			else {
-				message = getMessageBuilderFactory()
-						.withPayload(object)
-						.build();
-			}
-			if (isLoggingEnabled()) {
-				logger.debug(() -> "postReceive on channel '" + this + "', message: " + message);
-			}
-			return message;
+		Message<?> message = object instanceof Message<?> msg
+				? msg
+				: getMessageBuilderFactory().withPayload(object).build();
+		if (isLoggingEnabled()) {
+			logger.debug(() -> "postReceive on channel '" + this + "', message: " + message);
```

Comment:
Looks like we use a `LogAccessor` API here, so we don't need that `if (isLoggingEnabled()) {` since message supplier is going to be evaluated only if it is really enabled. See `LogAccessor` implementation.